elasticsearch/helpers/vectorstore/_async/embedding_service.py

```diff
@@ -0,0 +1,97 @@
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from elasticsearch import AsyncElasticsearch
+
+
+class AsyncEmbeddingService(ABC):
+    @abstractmethod
+    async def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        """Generate embeddings for a list of documents.
+
+        Args:
+            texts: A list of document strings to generate embeddings for.
+
+        Returns:
+            A list of embeddings, one for each document in the input.
+        """
+
+    @abstractmethod
+    async def embed_query(self, query: str) -> List[float]:
+        """Generate an embedding for a single query text.
+
+        Args:
+            text: The query text to generate an embedding for.
+
+        Returns:
+            The embedding for the input query text.
+        """
+
+
+class AsyncElasticsearchEmbeddings(AsyncEmbeddingService):
+    """Elasticsearch as a service for embedding model inference.
+
+    You need to have an embedding model downloaded and deployed in Elasticsearch:
+    - https://www.elastic.co/guide/en/elasticsearch/reference/current/infer-trained-model.html
+    - https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-deploy-models.html
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        es_client: AsyncElasticsearch,
+        user_agent: str,
+        model_id: str,
+        input_field: str = "text_field",
+        num_dimensions: Optional[int] = None,
```

Comment:
Is `num_dimensions` used anywhere else? Because in this module it is stored but not used for anything, right?

Reply:
Good point, this is a left-over from when I had the number of dimensions be part of the embedding service, which proved to create some problems. Cleaning up.